zerver/models.py

```diff
@@ -3313,6 +3314,12 @@ def increment_times_used(self) -> None:
         self.save(update_fields=["times_used"])
 
 
+class MissedMessageEmailBatch(models.Model):
+    user_profile: UserProfile = models.ForeignKey(UserProfile, on_delete=CASCADE, db_index=True)
+    next_process_time: datetime.datetime = models.DateTimeField(default=timezone_now)
+    events: List[object] = ArrayField(base_field=models.JSONField(), default=list, blank=True)
```

Comment:
See https://github.com/zulip/zulip/pull/18146#issuecomment-831013597; I think it's possible we want to avoid `ArrayField[JsonField]`.

Reply:
I didn't understand how the stackoverflow post is a concern for us, since we aren't searching inside the ArrayField column (and we won't be doing that event after having configurable time periods)

I pushed the `JSONField`-only version to save a round of review in case I'm missing something. It isn't too much work to go back to `ArrayField` either.

Reply:
Yeah, maybe it isn't; I didn't really remember the context.